### PR TITLE
Fix CTRL-C hanging if session is paused

### PR DESCRIPTION
**Pull request by xacrimon**
This was caused by our close subsystem only starting when the session was launched (i.e when required partipants had joined) meaning it was inactive before then. This PR moves that logic to the session initiation. I'm not sure when this regressed but it seems pretty recently as this worked when I demoed moderated sessions not too long ago.

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

**Review by Joerger — COMMENTED** [review]
Can you add a test for this actually? It'd be good to start getting some moderated sessions tests in `sess_test.go` if possible.

File: lib/srv/sess.go [hunk 1]
```diff
@@ -1092,6 +1060,39 @@ func (s *session) startInteractive(ch ssh.Channel, ctx *ServerContext, tempUser
...
+	// wait for exec.Cmd (or receipt of "exit-status" for a forwarding node),
+	// once it is received wait for the io.Copy above to finish, then broadcast
+	// the "exit-status" to the client.
+	go func() {
+		result, err := s.term.Wait()
+		if err != nil {
+			ctx.Errorf("Received error waiting for the interactive session %v to finish: %v.", s.id, err)
+		}
+
+		// wait for copying from the pty to be complete or a timeout before
```
Comment (zmb3) [inline lib/srv/sess.go:1073]: If we got an error above should we still be waiting here?

File: lib/srv/sess_test.go [hunk 2]
```diff
@@ -374,6 +375,48 @@ func TestInteractiveSession(t *testing.T) {
...
+				Kinds:  []string{string(types.SSHSessionKind)},
+				Modes:  []string{string(types.SessionPeerMode)},
+				Count:  999,
+			}},
+		},
+	})
+	require.NoError(t, err)
+
+	roles := services.NewRoleSet(role)
+	t.Run("Stop", func(t *testing.T) {
```
Comment (zmb3) [inline lib/srv/sess_test.go:405]: Does this need to be a subtest? Without the subtest, the parent test isn't really testing anything. This suggests to me that the code in the subtest is actually just part of the test and not a separate test.

File: lib/srv/sess.go (hunk 1 above)
Comment (xacrimon) [reply lib/srv/sess.go:1073]: I think so, wait can fail for a number of reasons but at least some of them still leave the terminal data valid so I think it makes sense to ensure it's all been copied.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**